src/infra/core/src/workspace_layout.rs

```diff
@@ -106,3 +117,59 @@ impl std::fmt::Display for WorkspaceVersion {
         write!(f, "{}", value)
     }
 }
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+const WORKSPACE_CONFIG_VERSION: i32 = 1;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct WorkspaceConfig {
```

Comment:
Why did you decide to use a separate config file instead of the existing config?

Reply:
As it's not a responsibility of the CLI layer. `WorkspaceLayout` is not a truly CLI concept on today, as it lives in the `infra` part. I cannot inject CLI-specific abstractions in the `infra` part.

Reply:
Ideally, it should be living in a new local-fs adapter, that is not referenced by the tests directly.

Reply:
Take a look at how we configure everything else on infra layer (container-runtime, ipfs gateway, engines etc.). The configuration lives in `CLIConfig` which during initialization is parsed and deconstructed into infra-specific config objects.

Is there any reason why we cannot do the same here and store tenancy as something like:
```yaml
workspace:
  tenancy: "multi-tenant"
```
in our main config?

Reply:
Two reasons:

1) I does not feel architecturally correct to me. There is no strict == between being in CLI mode and having a local FS workspace layout. 

2) Technically, I need to know if workspace is multi-tenant before DI catalog is started to get filled, as it has to be explicitly injected via builder arguments into several services. At least on today, reading `CLIConfig` happens at the very end of dependencies initialization. Changing the order of initialization feels unmotivated to me.

Reply:
Re. `2.` I think it's a matter of time when we need to change the DI catalog composition based on the configuration. I don't see anything that prevents us now from reading the config first, and it seems more logical to me. Current order just kinda "happened" rather than was designed.

Re. `1.` You can say this about every config option: IPFS  != container runtime != engine configuration. But this doesn't mean we should put each in a different config file, does it? If you follow same separation of:
- there is a config DTO in `infra` layer (agnostic of where values actually come from)
- there is a config section in `CLIConfig`
- we read CLI config and convert it into infra DTO

... then CLI config is purely "presentation" layer concern - just like any CLI flag or an argument. In fact coupling core tenancy configuration with file format and serialization feels less architecturally sound to me.

It also:
- Complicates workspace structure
- And any improvements we make to `CLIConfig` (e.g. allowing to override config via ENV vars or CLI args) will not benefit the tenancy config...

Reply:
Re 3. Config is modifiable via commands. Workspace multi-tenance property should be readonly.

Reply:
How about we continue with the current (separate file) solution, but modify `WorkspaceLayout::create()` not to write a config file for a single-tenant workspace, so that in prod cases this config was not present?

Reply:
Done.

```
        // Only save the workspace configuration if it is different from default
        let ws_config = WorkspaceConfig::new(multi_tenant);
        if ws_config != WorkspaceConfig::default() {
            ws_config.save_to(&ws.config_path).int_err()?;
        }
```